central/complianceoperator/v2/scanconfigurations/datastore/datastore_impl.go

```diff
@@ -138,50 +136,50 @@ func (ds *datastoreImpl) UpdateClusterStatus(ctx context.Context, scanID string,
 		return errors.Errorf("could not pull config for cluster %q because it does not exist", clusterID)
 	}
 
-	ds.keyedMutex.Lock(scanID)
-	defer ds.keyedMutex.Unlock(scanID)
+	ds.keyedMutex.Lock(scanConfigID)
+	defer ds.keyedMutex.Unlock(scanConfigID)
 
 	// Ensure the scan configuration exists
-	_, found, err := ds.GetScanConfiguration(ctx, scanID)
+	_, found, err := ds.GetScanConfiguration(ctx, scanConfigID)
 	if err != nil || !found {
-		return errors.Errorf("Unable to find scan configuration id %q", scanID)
+		return errors.Errorf("Unable to find scan configuration id %q", scanConfigID)
 	}
 
 	// Need to build a deterministic ID from clusterID and scanID to ensure we always have the latest status
 	clusterUUID, err := uuid.FromString(clusterID)
 	if err != nil {
-		return errors.Wrapf(err, "Unable to build scan configuration status id based off %q", scanID)
+		return errors.Wrapf(err, "Unable to build scan configuration status id based off %q", scanConfigID)
 	}
-	statusKey := uuid.NewV5(clusterUUID, scanID).String()
+	statusKey := uuid.NewV5(clusterUUID, scanConfigID).String()
 
 	clusterScanStatus := &storage.ComplianceOperatorClusterScanConfigStatus{
-		Id:          statusKey,
-		ClusterId:   clusterID,
-		ClusterName: cluster.GetName(),
-		ScanId:      scanID,
-		Errors:      []string{clusterStatus},
+		Id:           statusKey,
+		ClusterId:    clusterID,
+		ClusterName:  cluster.GetName(),
+		ScanConfigId: scanConfigID,
+		Errors:       []string{clusterStatus},
 	}
 
 	return ds.statusStorage.Upsert(ctx, clusterScanStatus)
 }
 
 // GetScanConfigClusterStatus retrieves the scan configurations status per cluster specified by scan id
-func (ds *datastoreImpl) GetScanConfigClusterStatus(ctx context.Context, scanID string) ([]*storage.ComplianceOperatorClusterScanConfigStatus, error) {
-	if ok, err := scanConfigurationsSAC.ReadAllowed(ctx); err != nil {
-		return nil, err
-	} else if !ok {
-		return nil, nil
-	}
-
+func (ds *datastoreImpl) GetScanConfigClusterStatus(ctx context.Context, scanConfigID string) ([]*storage.ComplianceOperatorClusterScanConfigStatus, error) {
 	return ds.statusStorage.GetByQuery(ctx, search.NewQueryBuilder().
-		AddExactMatches(search.ComplianceOperatorScanConfig, scanID).ProtoQuery())
+		AddExactMatches(search.ComplianceOperatorScanConfig, scanConfigID).ProtoQuery())
```

Comment:
Technically it probably should however at this time, the only path to here is when building the results to send back.  Meaning the get configuration methods have already filtered out those without access.

That said, since we are likely to change GetScanConfiguration to best effort then this one will not need updated as the store would do best effort by default.